src/manifold/src/smoothing.cpp

```diff
@@ -53,131 +53,14 @@ void FillRetainedVerts(Vec<Barycentric>& vertBary,
-          u * vertPos[edgeVerts.first] + v * vertPos[edgeVerts.second];
-
-      const int tri = edgeVerts.halfedgeIdx / 3;
-      const int idx = edgeVerts.halfedgeIdx - 3 * tri;
-      glm::vec3 uvw(0);
-      uvw[idx] = u;
-      uvw[Next3(idx)] = v;
-      vertBary[vert] = {tri, uvw};
-    }
-  }
-};
-
-struct InteriorVerts {
-  VecView<glm::vec3> vertPos;
-  VecView<Barycentric> vertBary;
-  const int startIdx;
-  const int n;
-  VecView<const Halfedge> halfedge;
-
-  void operator()(int tri) {
-    const float invTotal = 1.0f / n;
-    int pos = startIdx + tri * VertsPerTri(n - 2);
-    for (int i = 0; i <= n; ++i) {
-      for (int j = 0; j <= n - i; ++j) {
-        const int k = n - i - j;
-        const float u = invTotal * j;
-        const float v = invTotal * k;
-        const float w = invTotal * i;
-        if (i == 0 || j == 0 || k == 0 || j == n - i) continue;
-
-        vertPos[pos] = u * vertPos[halfedge[3 * tri].startVert] +      //
-                       v * vertPos[halfedge[3 * tri + 1].startVert] +  //
-                       w * vertPos[halfedge[3 * tri + 2].startVert];
-
-        vertBary[pos++] = {tri, {u, v, w}};
-      }
-    }
-  }
-};
-
-struct SplitTris {
-  VecView<glm::ivec3> triVerts;
-  VecView<Halfedge> halfedge;
-  VecView<const int> half2Edge;
-  const int edgeIdx;
-  const int triIdx;
-  const int n;
-
-  int EdgeVert(int i, int inHalfedge) const {
-    bool forward = halfedge[inHalfedge].IsForward();
-    int edge = forward ? half2Edge[inHalfedge]
-                       : half2Edge[halfedge[inHalfedge].pairedHalfedge];
-    return edgeIdx + (n - 1) * edge + (forward ? i - 1 : n - 1 - i);
-  }
-
-  int TriVert(int i, int j, int tri) const {
-    --i;
-    --j;
-    int m = n - 2;
-    int vertsPerTri = (m * m + m) / 2;
-    int vertOffset = (i * (2 * m - i + 1)) / 2 + j;
-    return triIdx + vertsPerTri * tri + vertOffset;
-  }
-
-  int Vert(int i, int j, int tri) const {
-    bool edge0 = i == 0;
-    bool edge1 = j == 0;
-    bool edge2 = j == n - i;
-    if (edge0) {
-      if (edge1)
-        return halfedge[3 * tri + 1].startVert;
-      else if (edge2)
-        return halfedge[3 * tri].startVert;
-      else
-        return EdgeVert(n - j, 3 * tri);
-    } else if (edge1) {
-      if (edge2)
-        return halfedge[3 * tri + 2].startVert;
-      else
-        return EdgeVert(i, 3 * tri + 1);
-    } else if (edge2)
-      return EdgeVert(j, 3 * tri + 2);
-    else
-      return TriVert(i, j, tri);
-  }
-
-  void operator()(int tri) {
-    int pos = n * n * tri;
-    for (int i = 0; i < n; ++i) {
-      for (int j = 0; j < n - i; ++j) {
-        int a = Vert(i, j, tri);
-        int b = Vert(i + 1, j, tri);
-        int c = Vert(i, j + 1, tri);
-        triVerts[pos++] = glm::ivec3(c, a, b);
-        if (j < n - 1 - i) {
-          int d = Vert(i + 1, j + 1, tri);
-          triVerts[pos++] = glm::ivec3(b, d, c);
-        }
-      }
-    }
+    half2Edge[halfedges[halfedge].pairedHalfedge] = edge;
```

Comment:
is this a bug fix or needed due to the new feature?

Reply:
I used to only need that mapping for the forward halfedges, but now I need it for both.